File: 657-robot-return-to-origin/657-robot-return-to-origin.java
class Solution {
    public boolean judgeCircle(String moves) {
        int n = 0;
        int m = 0;
        for(int i = 0; i<moves.length(); i++){
            if(moves.charAt(i) == 'R'){
                n++;
            }
            if(moves.charAt(i) == 'L'){
                n--;
            }
            if(moves.charAt(i)== 'U'){
                m++;
            }
            if(moves.charAt(i) == 'D'){
                m--;
            }
        }
        if(m == 0 & n ==0){
            return true;
        }else{
            return false;
        }
    }
}

File: 657-robot-return-to-origin/657-robot-return-to-origin.js
/**
 * @param {string} moves
 * @return {boolean}
 */
var judgeCircle = function(moves) {
    let updown = 0;
    let leftright = 0;
    for(let move of moves){
        if(move === "U"){
            updown++;
        }else if(move === "D"){
            updown--;
        }else if(move === "L"){
            leftright--;
        }else if(move === "R"){
            leftright++;
        }
    }
    return updown === 0 && leftright === 0;
};